src/tools/rbd/action/MergeDiff.cc

```diff
@@ -376,7 +376,7 @@ static int do_merge_diff(const char *first, const char *second,
       s_len = 0;
       s_tag = 0;
     } else {
-      ceph_assert(f_end && s_end);
+      ceph_assert(f_end) && ceph_assert(s_end);
```

Comment:
This doesn't compile because `ceph_assert` is a macro and doesn't return anything.  If one really wanted to break this assert into two separate asserts, the equivalent would be:
```
ceph_assert(f_end);
ceph_assert(s_end);
```
But then I don't see the point of this change -- `f_end && s_end` is really not a complex condition.

Reply:
Hi! this is a change from grace hopper open source day, i'm new to the project, the issue link is [here](https://trello.com/c/Ri6ixQ5X/615-osd-cleanup-multi-condition-asserts), do you think it'll make more sense to break multiple asserts in other more complex conditions? Thanks!

Reply:
@tchaikov previously raised some concerns about `ceph_assert` not being completely free (potential executable size bloat, etc).  However the upside is better telemetry reports, so it's a bit of a balancing act.  I would suggest approaching these on a case by case basis:

- trello card mentions the OSD code specifically, this is mainly `src/osd` subdirectory
- pick a few of the really complex conditions there (`A && B` or `A || B` probably don't qualify unless `A` and/or `B` is itself complex)
- actually compile and test your change

Reply:
@li3317 I know you had trouble setting up the project to compile-- let me check around in the OSD code for a better case.

Reply:
Also @idryomov for context, the Open Source day participants had very limited time to set up the correct environment in order to compile. This is why @li3317 and a few others had to test their changes on make check instead of compiling locally.

Reply:
@idryomov @li3317 this one could be worth breaking up, as the ceph_assert is already written on two lines:

https://github.com/ceph/ceph/blob/a79e029a727d594c67f1beb71ebeeb33f744583d/src/osd/ECBackend.cc#L2356-L2357

What do you guys think?

Reply:
Also @li3317 if you need to go through with another change, I can provide you with the git commits you'd need to fix your PR. But first let's see what @idryomov thinks. :)

Reply:
This one certainly seems less straightforward and therefore a better candidate, but this being OSD code it's up to @rzarzynski and @neha-ojha, not me ;)

Reply:
@idryomov makes sense!

@li3317 if you're still interested in making the above change, let me know and I can help you with the git commands if needed.

Reply:
Hi @ljflores yes no problem I can make that change, and the git commands would be great thank you! I was finally able get the vm running but still have trouble setting some things up so I'm still not sure if I can compile locally though.

Reply:
@li3317 no worries! If you have any issues setting up your vstart environment ( which is required to compile ) let me know and we can work through that too. For now, you can continue pushing new changes here to see whether it compiles on make check, and then fix it accordingly.

So what you'll need to do is:
1. Go to your terminal
2. `cd ceph`
3. `git checkout ghc-osd-li3317`
4. `git log`
5. Copy the SHA of the commit right *before* `tools/rbd/action: break multi condition asserts into multiple asserts` (the long line of letters and numbers right after the word "commit"

So if `git log` looks like this for you:
```
commit e78edb2b16672dfcc0ed648db0ded9a086d895e3 (HEAD -> ghc-osd-li3317, li3317/ghc-osd-li3317)
Author: li3317 <li3317@purdue.edu>
Date:   Fri Sep 16 15:45:30 2022 -0400

    tools/rbd/action: break multi condition asserts into multiple asserts
    
    Signed-off-by: Annabel Li <li3317@purdue.edu>

commit e4638a14ec4f27dd27f3e0707e7b03901e99813a (li3317/main)
Merge: ab52b8cc71c 0d6100bbf99
Author: Nizamudeen A <nia@redhat.com>
Date:   Fri Sep 16 20:49:48 2022 +0530

    Merge pull request #47849 from torchiaf/fix/37327
    
    mgr/dashboard: Add details to the modal which displays the `safe-to-d…
    
    Reviewed-by: Nizamudeen A <nia@redhat.com>
    Reviewed-by: Volker Theile <vtheile@suse.com>
```
You'll want to copy this SHA: `e4638a14ec4f27dd27f3e0707e7b03901e99813a`.

6. `git rebase -i <the SHA you copied`
7. Replace the word `pick` with `drop` since we want to get rid of this commit.
8. Save the commit (in VIM, this is just `:wq`.
9. Running `git log`, you can see that your branch is based on the main branch again, and we have gotten rid of the commit we don't want.

10. Next, you'll go to the file in https://github.com/ceph/ceph/pull/48150#discussion_r974752474 and make the change I described there.
11. Run `git diff` to make sure that your change looks as expected.
12. `git add <the file that you changed>`
13. `git commit`
14. This will bring you to a text file where you can write your commit. You can use the same structure as https://github.com/ceph/ceph/pull/48150/commits/e78edb2b16672dfcc0ed648db0ded9a086d895e3, except change the filepath from `tools/rbd/action` to `osd`, since that's where ECBackend.cc is located.
15. Don't forget to sign off your commit!
16. Save the commit (`:wq` on VIM)
17. Now, since we've re-written the history of the branch, we will have to force push the changes: `git push -f <the name of your forked repository> ghc-osd-li3317`
18. If you're unsure of the name of your forked repository, you can run `git remote -v` and see which name corresponds with https://github.com/li3317/ceph.git.

Then you're done! You've successfully updated your branch with the new commit.
When you're done, you can ping me (@ljflores) and let me know that it needs another review. And you can monitor make check (under the tests below) to see if your code compiles.

Reply:
Thank you so much @ljflores ! I've just pushed the change and I'll for sure save the list of git commands for future!